rocketbot/commands/dms.py

```diff
@@ -0,0 +1,55 @@
+from typing import List, Tuple
+import subprocess
+import os
+import re
+
+import dmsclient as dms
+
+import rocketbot.commands as c
+import rocketbot.master as master
+import rocketbot.models as m
+
+
+class DMS(c.BaseCommand):
+    def __init__(self, master: master.Master, token: str):
+        self._create_dmsclient_config_if_missing(token)
+        self.master = master
+
+    def usage(self) -> List[Tuple[str, str]]:
+        return [
+            ('order', 'Orders product in dms for yourself.'),
+        ]
+
+    def can_handle(self, command: str) -> bool:
+        """Check whether the command is applicable
+        """
+        return command in ['dms', 'drink', 'drinks', 'order']
+
+    async def handle(self, command: str, args: str, message: m.Message) -> None:
+        """Handle the incoming message
+        """
+        argv = re.split(r'\s+', args.strip())
+
+        userargv = [arg for arg in argv if arg.startswith('-u') or arg.startswith('--user')]
+        if len(userargv) == 0:
+            userid = message.u.username
+            argv.append('--user={}'.format(userid))
+        if '--force' not in argv:
+            argv.append('--force')
+
+        if command == 'order':
+            argv = [command, *argv]
+
+        if command in ('dms', 'drink', 'drinks'):
+            dms_result = subprocess.run(['dms', *argv],
```

Comment:
This should not block the main loop of the bot because the surrounding 'handle' function is async.

Reply:
Since the subprocess is spawned in a syncronous way the controlflow is never given back to the event loop und therefore its blocking. Would be different if there was something like 
```suggestion
            dms_result = await subprocess.run_async(['dms', *argv],
```